app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+
+  validates :name, presence: { message: 'Name field can not left blank' }
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  private
+
+  def most_recent_posts
+    posts.order('created_at DESC').limit(3)
+  end
```

Comment:
- [ ] Kindly write unit tests for this method. I would advise you to make this a _**public**_ method. So that you write unit tests for this method. Remember that if you make a method **_private_**, you will not be able to call that method from outside the class where it was declared. 👍